pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -247,10 +265,19 @@ func (g *Gatherer) updateAlertsCache(ctx context.Context) error {
 		return err
 	}
 
-	return g.updateAlertsCacheFromClient(ctx, metricsClient)
+	if err := g.updateAlertsCache(ctx, metricsClient); err != nil { //nolint:govet
+		return err
+	}
+
+	configClient, err := configv1client.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return err
+	}
+
+	return g.updateVersionCache(ctx, configClient)
```

Comment:
I guess this logic here basically means that if the `updateAlertsCache` fails (we couldn't connect to the metrics for some reason) then the `updateVersionCache` is not executed. Is that right? Does it mean that the conditions for cluster version will not be evaluated when we can't connect to the metrics?

Reply:
> I guess this logic here basically means that if the updateAlertsCache fails (we couldn't connect to the metrics for some reason) then the updateVersionCache is not executed. Is that right?

Yes, but there's no point to update version cache if we don't have valid alerts cache since current rules contain alerts anyway. 

> Does it mean that the conditions for cluster version will not be evaluated when we can't connect to the metrics?

Yup, conditional gatherer won't do it's job if it can't initialize any of the caches. We'd get the error in the metadata tho.

Reply:
> Yes, but there's no point to update version cache if we don't have valid alerts cache since current rules contain alerts anyway.
I am not sure I follow. We don't have any conditional rules with version yet. I think it's difficult to predict if they will have any alerts as well. If they will then it probably makes sense, but if they will not then it seems as unnecessary limitation.

Reply:
Updated changing the logic little bit so we would get more info in the archive

Reply:
Thanks